### PR TITLE
Add a tutorial example for the triaxialDriver

### PR DESCRIPTION
**Pull request by jhuang2601**
This PR documents the application of the triaxial Driver.

As the triaxial Driver is covered by the unit test, smoke test and rebaseline of the integratedTest are not required for this PR.

### CODE REVIEW

**Review by herve-gross — APPROVED** [review]
This is very useful, thank you Jian!

File: inputFiles/solidMechanics/triaixalDriverTables/axialStrain.geos [hunk 1]
```diff
@@ -0,0 +1,6 @@
+0.0
```
Comment (joshua-white) [inline inputFiles/solidMechanics/triaixalDriverTables/axialStrain.geos:1]: i would actually put these files in something like`inputFiles/triaxialDriver/tables`.  We organize mainly by physics solver, and the driver is unrelated to the LagrangianFEM examples already in `solidMechanics`.  Also, note you have a typo in the path: `triaixial`

File: src/docs/sphinx/basicExamples/triaxialDriver/Example.rst [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+Then, ``steps="200"`` defines the number of load steps and ``output="simulationResults.txt"`` specifies an output file to which the simulation results will be written. 
+
+
+.. literalinclude:: ../../../../../inputFiles/solidMechanics/triaxialDriver_ExtendedDruckerPrager.xml
+    :language: xml
+    :start-after: <!-- SPHINX_TASK -->
+    :end-before: <!-- SPHINX_TASK_END -->
+
+
+In addition to triaxial tests, volumetric and oedometer tests can be simulated by changing the ``strainControl`` mode, and by defining loading controls in axial and radial direction accordingly. A volumetric test can be modelled by setting the axial and radial control functions to the same strain function, whereas an oedometer test runs by setting the radial strain to zero.
```
Comment (joshua-white) [inline src/docs/sphinx/basicExamples/triaxialDriver/Example.rst:101]: Maybe link to triaxialDriver documentation here

File: src/docs/sphinx/basicExamples/triaxialDriver/Example.rst [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+All constitutive parameters such as density, viscosity, bulk modulus, and shear modulus are specified in the International System of Units.
+
+
+------------------------------------------------------------------
+Functions
+------------------------------------------------------------------
+
+The ``TriaxialDriver`` uses a simple form of time-stepping to advance through the loading steps, allowing for simulating both rate-dependent and rate-independent models.  
+
+In this case, users should define two different time history functions (``strainFunction`` and ``stressFunction``) to describe loading conditions in axial and radial direction respectively. More specifically, the table functions in this example include the temporal variations of radial stress and axial strain, which rely upon the external files in the table directory.
```
Comment (joshua-white) [inline src/docs/sphinx/basicExamples/triaxialDriver/Example.rst:132]: Link the Function documentation, and also note that simple functions can be embedded directly in the XML.  In fact, in this case, I would recommend it.  The external files are overkill.

File: src/docs/sphinx/basicExamples/triaxialDriver/Example.rst [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+  2.5000e-02 -1.0000e-04 2.5000e-05 2.5000e-05 -1.1500e+07 -1.0000e+07 -1.0000e+07 1.0000e+00 0.0000e+00 
+  5.0000e-02 -2.0000e-04 5.0000e-05 5.0000e-05 -1.3000e+07 -1.0000e+07 -1.0000e+07 1.0000e+00 0.0000e+00 
+  7.5000e-02 -3.0000e-04 7.5000e-05 7.5000e-05 -1.4500e+07 -1.0000e+07 -1.0000e+07 1.0000e+00 0.0000e+00 
+  1.0000e-01 -4.0000e-04 1.0000e-04 1.0000e-04 -1.6000e+07 -1.0000e+07 -1.0000e+07 1.0000e+00 0.0000e+00 
+  ...
+
+Note that the file contains two columns for radial strains (``radial_strain_1`` and ``radial_strain_2``) and two columns for radial stresses (``radial_stress_1`` and ``radial_stress_2``). For isotropic materials, the stresses and strains along the two radial axes would be the same. However, the stresses and strains in the radial directions can differ for cases with anisotropic materials and true-triaxial loading conditions.
+
+This output file can be processed and visualized using any tools. As an example here, with the provided Python script, the simulated stress-strain curve, p-q diagram and relationship between volumetric strain and axial strain are plotted, and used to validate results against experimental observations:
```
Comment (joshua-white) [inline src/docs/sphinx/basicExamples/triaxialDriver/Example.rst:250]:
```suggestion
This output file can be processed and visualized using any tool. As an example here, with the provided python script, the simulated stress-strain curve, p-q diagram and relationship between volumetric strain and axial strain are plotted, and used to validate results against experimental observations:
```

**Review by joshua-white — APPROVED** [review]
Can you make sure this example is invoked when we run the ctest baseline comparisons as we do for the other triax test XMLs?  We need to test it or else it may get out of date.

Looks great!

File: src/docs/sphinx/basicExamples/triaxialDriver/Example.rst (hunk 2 above)
Comment (jhuang2601) [reply src/docs/sphinx/basicExamples/triaxialDriver/Example.rst]: Add it

File: src/docs/sphinx/basicExamples/triaxialDriver/Example.rst (hunk 3 above)
Comment (jhuang2601) [reply src/docs/sphinx/basicExamples/triaxialDriver/Example.rst]: Add the suggested documentation and note. For the cases with complex loading conditions (such as, combination of triaxial and volumetric compression) , external tables need to be turned on for importing lab controls, which might be frequently used by the users. Therefore, I choose this case to cover this function.

**Comment by jhuang2601** [comment]
> Can you make sure this example is invoked when we run the ctest baseline comparisons as we do for the other triax test XMLs? We need to test it or else it may get out of date.
> 
> Looks great!

As discussed, we will revisit the suggested option for this example, when a better strategy for unitTest and integratedTest is defined.

**Pull request merged**